nncf/openvino/graph/layer_attributes.py

```diff
@@ -68,8 +70,84 @@ def get_const_port_ids(self) -> List[int]:
         return []
 
 
+def get_conv_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param node: Target convolution node.
+    :return: Target convolution Node weights layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    return get_conv_weights_layout(
+        ov_metatype=node.metatype, weights_shape=layer_attributes.constant_attributes[port_id]["shape"]
+    )
+
+
+def get_linear_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target linear node.
+
+    :param node: Target linear node.
+    :return: Target linear Node weight layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    constant_layer_attrs = layer_attributes.constant_attributes[port_id]
+    return get_linear_weights_layout(
+        weights_shape=constant_layer_attrs["shape"],
+        transpose=constant_layer_attrs["transpose"],
+        port_id=port_id,
+    )
+
+
+def _get_constant_port_id_from_layer_attributes(layer_attributes: OVLayerAttributes) -> int:
+    """
+    Returns constant ports id for convolutional and linear ops layer attributes.
+
+    :param layer_attributes: Target convolutional/linear layer op layer attributes.
+    :return: Constant port id for the target convolutional/linear model.
+    """
+    port_ids = list(layer_attributes.constant_attributes.keys())
+    assert len(port_ids) == 1
+    return port_ids[0]
+
+
+def get_conv_weights_layout(ov_metatype: OVOpMetatype, weights_shape: Tuple[int, ...]) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param ov_metatype: Target convolution node OpenVINO metatype.
+    :param weights_shape: Shape of the target convolution node weight.
+    :return: Target convolution node weights layout.
+    """
+    weights_layout = ov_metatype.const_layout
+    kernel_size = weights_shape[len(weights_layout) :]
+    weights_layout += [OVLayoutElem.SPATIAL] * len(kernel_size)
```

Comment:
Because layout depends on kernel size:
Conv 1D: [C_OUT, C_IN, X]
Conv 2D: [C_OUT, C_IN, X, Y]
Conv 3D: [C_OUT, C_IN, X, Y, Z] and etc.
But we have only one metatype for all cases: `OVConvolutionMetatype`. Is there any way we can calculate layout inside conv metatypes I don't aware of?

Reply:
Essentially that's what I mean. You introduced the abstraction that can not fully specify the constant layout of `OVConvolutionMetatype`. However, you just need C_OUT, C_IN and GROUPS dimensions.